### PR TITLE
Conformance test

### PR DESCRIPTION
**Pull request by Rdk-T**
Test the slave stack on a custom slave with Beckhoff's Conformance Test Tool.

~150 tests pass
~270 tests skipped (no mailbox supported)

all_test_pass

### CODE REVIEW

File: examples/slave/nuttx/xmc4800/main_relax.cc [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (leducp) [inline examples/slave/nuttx/xmc4800/main_relax.cc:1]: line feed

File: examples/slave/nuttx/xmc4800/main_relax.cc [hunk 2]
```diff
@@ -6,7 +7,7 @@
 int main(int , char *[])
```
Comment (leducp) [inline examples/slave/nuttx/xmc4800/main_relax.cc:7]: comma space (before)

File: examples/slave/nuttx/xmc4800/main_foot.cc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        uint16_t boardStatus;
+    } __attribute__((packed));
+
+    struct Output
+    {
+        uint16_t watchdog_counter;
+    } __attribute__((packed));
+}
+
+int main(int , char *[])
```
Comment (leducp) [inline examples/slave/nuttx/xmc4800/main_foot.cc:41]: comma space (before)

File: examples/slave/nuttx/xmc4800/main_relax.cc [hunk 4]
```diff
@@ -6,7 +7,7 @@
...
 {
     using namespace kickcat;
```
Comment (leducp) [inline examples/slave/nuttx/xmc4800/main_relax.cc:9]: line feed after (I feel it more readable like this, minor though)

File: lib/slave/include/kickcat/AbstractESC.h [hunk 5]
```diff
@@ -56,16 +57,28 @@ namespace kickcat
 
         uint16_t al_status() { return al_status_;};
 
-        private:
+        bool has_expired_watchdog() {return not (watchdog_ & 0x1);};
```
Comment (leducp) [inline lib/slave/include/kickcat/AbstractESC.h:60]: space before/after {}

remove ; at the end

File: lib/slave/src/AbstractESC.cc [hunk 6]
```diff
@@ -26,12 +26,35 @@ namespace kickcat
...
 
+    void AbstractESC::set_sm_activate(SyncManagerConfig const& sm_conf, bool is_activated)
+    {
+        auto create_sm_address = [](uint16_t reg, uint16_t sm_index)
+        {
+            return reg + sm_index * 8;
+        };
+
+        SyncManager sm;
+        reportError(read(create_sm_address(reg::SYNC_MANAGER, sm_conf.index), &sm, sizeof(sm)));
```
Comment (leducp) [inline lib/slave/src/AbstractESC.cc:45]: no return ? (instead of just a trace)

**Review by leducp — APPROVED** [review]

File: lib/slave/src/AbstractESC.cc (hunk 6 above)
Comment (Rdk-T) [reply lib/slave/src/AbstractESC.cc:45]: In the whole class the errors are just traced, I didn't use them to do meaningful things

**Pull request merged**